src/api/OrderAPI.ts

```diff
@@ -278,11 +269,41 @@ export class OrderAPI {
 
   /* ============ Private Assertions ============ */
 
+  private assertFillAmountValidity(
+    signedIssuanceOrder: SignedIssuanceOrder,
+    quantityToFill: BigNumber,
+    orders: ExchangeOrder[],
+  ): void {
+    let zeroExFillAmounts = SetProtocolUtils.CONSTANTS.ZERO;
+    _.each(orders, (order: any) => {
+      if (SetProtocolUtils.isZeroExOrder(order)) {
+        this.assert.common.greaterThanZero(
```

Comment:
To enforce the right layers of abstraction, can we create a separate function for 0x Order assertions?

Reply:
So you loop through all the orders and just check the 0x ones? 

Would it make more sense to loop once through all the orders in a `validateOrders` function which has a switch statement. The switch statement will route the order to the right order validator function and then return with a sum of whether its valid as well as the amount of maker token used